db/blob/db_blob_index_test.cc

```diff
@@ -154,7 +157,8 @@ TEST_F(DBBlobIndexTest, Write) {
 }
 
 // Get should be able to return blob index if is_blob_index is provided,
-// otherwise return Status::NotSupported status.
+// otherwise it should return Status::NotSupported (when reading from memtable)
+// or Status::Corruption (when reading from SST).
```

Comment:
On second thought, can you remind me why `Corruption` is returned if `is_blob_index == nullptr` and read from SST? I am thinking maybe `InvalidArgument` is a better fit?

Reply:
I should probably tweak that comment a bit more. :) Here's the deal: that's actually a test case related to the old `StackableDB` based BlobDB. Namely, `GetImplOptions::is_blob_index` is only non-null when the old BlobDB is used. Historically, ending up with a `BlobIndex` when `is_blob_index` is `nullptr` meant that the user had accidentally opened the base DB of a `BlobDB` instance (i.e. used `DB::Open` instead of `BlobDB::Open`), thus we returned `NotSupported` with a message instructing the user to use `BlobDB::Open` instead.

With the new integrated BlobDB in the picture, the above assumption no longer holds, that is, we can't really tell apart a) the above case of the application accidentally opening the base DB of a stacked BlobDB and b) a real corruption when using the new integrated BlobDB (an example would be a `BlobIndex` pointing to a blob file not in the `Version`). Since we have two pick one of the two, I went with the second one since I think the first case is extremely unlikely and the message would be pretty confusing for users.